script/utils/CurrentConfigCheck.s.sol

```diff
@@ -0,0 +1,570 @@
+        );
+        vm.stopBroadcast();
+    }
+
+    function simulateLegacyTimelockActions() public {
+        // TODO
+        // give proxy admin ownership to timelock controller
+        // eigenTokenProxyAdmin.transferOwnership(address(timelockController));
+
+        // swapOwner(address previousOwner, address oldOwner, address newOwner)
+        // TODO: figure out if this is the correct input for all chains or the communityMultisig is correct on some
+        address previousOwner = address(1);
+        // address previousOwner = communityMultisig;
+        bytes memory data_swapTimelockToProtocolTimelock =
+            abi.encodeWithSignature("swapOwner(address,address,address)", previousOwner, address(timelock), address(protocolTimelockController));
+        bytes memory callToExecutor = encodeForExecutor({
+            from: timelock,
+            to: address(executorMultisig),
+            value: 0,
+            data: data_swapTimelockToProtocolTimelock,
+            operation: ISafe.Operation.Call 
+        });
+        // TODO: get appropriate value for chain instead of hardcoding
+        // uint256 timelockEta = block.timestamp + timelock.delay();
+        uint256 timelockEta = block.timestamp + 10 days;
+        (bytes memory calldata_to_timelock_queuing_action, bytes memory calldata_to_timelock_executing_action) =
+            encodeForTimelock({
+                to: address(executorMultisig),
+                value: 0,
+                data: callToExecutor,
+                timelockEta: timelockEta
+            });
+        vm.startPrank(operationsMultisig);
+
+        (bool success, /*bytes memory returndata*/) = timelock.call(calldata_to_timelock_queuing_action);
+        require(success, "call to timelock queuing action 1 failed");
+
+        vm.warp(timelockEta);
+        (success, /*bytes memory returndata*/) = timelock.call(calldata_to_timelock_executing_action);
+        require(success, "call to timelock executing action 1 failed");
+
+        vm.stopPrank();
+    }
+
+    function simulateEigenTokenTimelockControllerActions() public {
+        vm.startPrank(foundationMultisig);
+
+        uint256 tx_array_length = 3;
+        address[] memory targets = new address[](tx_array_length);
+        uint256[] memory values = new uint256[](tx_array_length);
+        bytes[] memory payloads = new bytes[](tx_array_length);
+        // 1. transfer upgrade rights over EIGEN token from eigenTokenProxyAdmin to eigenLayerProxyAdmin
+        targets[0] = address(eigenTokenProxyAdmin);
+        payloads[0] = abi.encodeWithSelector(ProxyAdmin.changeProxyAdmin.selector, address(EIGEN), address(eigenLayerProxyAdmin));
+        // 2. transfer ownership of EIGEN token to executorMultisig
+        targets[1] = address(EIGEN);
+        payloads[1] = abi.encodeWithSelector(Ownable.transferOwnership.selector, executorMultisig);
+        // 3. transfer ownership of bEIGEN token to executorMultisig
+        targets[2] = address(bEIGEN);
```

Comment:
move to other function?

Reply:
no? why? this timelock is currently the owner of both EIGEN and bEIGEN, so this is how we will actually perform this action.